mir-ci/mir_ci/tests/robot/platforms/vnc/Vnc.py

```diff
@@ -0,0 +1,211 @@
+        self.ROBOT_LIBRARY_LISTENER = self
+        self._client: asyncvnc.Client = None
+        self._host = host
+        self._port = port
+        self._username = username
+        self._password = password
+        self._pointer_position: Optional[Tuple[int, int]] = None
+        self._rpa_images = Images()
+
+    @keyword
+    async def match(self, template: str, timeout: int = 5) -> List[dict]:
+        """
+        Grab screenshots and compare until there's a match with the provided
+        template.
+
+        :param template: path to an image file to be used as template
+        :param timeout: timeout in seconds
+        :return: list of matched regions
+        :raises ImageNotFoundError: if no match is found within the timeout
+        """
+        await self.connect()
+        regions = []
+        end_time = time.time() + float(timeout)
+        screenshot = None
+        while time.time() <= end_time:
+            screenshot = Image.fromarray(await asyncio.wait_for(self._client.screenshot(), timeout))
+            try:
+                regions = self._rpa_images.find_template_in_image(
+                    screenshot,
+                    template,
+                    tolerance=self.TOLERANCE,
+                )
+            except (RuntimeError, ValueError, ImageNotFoundError):
+                continue
+            else:
+                break
+        else:
+            if screenshot:
+                self._log_failed_match(screenshot, template)
+            raise ImageNotFoundError
+
+        return [
+            {
+                "left": region.left,
+                "top": region.top,
+                "right": region.right,
+                "bottom": region.bottom,
+            }
+            for region in regions
+        ]
+
+    @keyword
+    async def type_string(self, text: str):
+        await self.connect()
+        self._client.keyboard.write(text)
+
+    @keyword
+    async def move_pointer_to_absolute(self, x: int, y: int) -> None:
+        await self.connect()
+        self._client.mouse.move(x, y)
+        self._pointer_position = (x, y)
+
+    @keyword
+    async def move_pointer_to_proportional(self, x: float, y: float) -> tuple[int, int]:
+        await self.connect()
+        absolute_position = self.get_absolute_from_proportional(x, y)
+        self.move_pointer_to_absolute(absolute_position[0], absolute_position[1])
+        return absolute_position
+
+    @keyword
+    async def walk_pointer_to_absolute(self, x: int, y: int, step_distance: int, delay: float) -> None:
+        await self.connect()
+        assert self._pointer_position is not None, "Cannot walk without moving the pointer at least once"
+        assert step_distance > 0, "Step distance must be positive"
+
+        distance = ((self._pointer_position[0] - x) ** 2 + (self._pointer_position[1] - y) ** 2) ** 0.5
+        if distance == 0:
+            return
```

Comment:
This might be worth logging a warning, WDYT?

Reply:
Yes, I agree.

Reply:
Oh, I've looked at the code, and the other backends (WaylandHid and ZapperHid) don't log any warnings in this case. I think we should stick to this behavior for consistency, unless we change all backends.

Reply:
If that's the existing behavior, I am okay keeping it